flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,190 @@ import Sentry
+                options.sessionTrackingIntervalMillis = sessionTrackingIntervalMillis
+            }
+
+            if let dist = arguments["dist"] as? String{
+                options.dist = dist
+            }
+
+            if let integrations = arguments["integrations"] as? [String]{
+                options.integrations = integrations
+            }
+            if let maxBreadcrumbs = arguments["maxBreadcrumbs"] as? UInt{
+                options.maxBreadcrumbs = maxBreadcrumbs
+            }
+
+            /*
+             missing fields : enableAutoNativeBreadcrumbs, diagnostic level,
+             enableNativeCrashHandling,platform, web, packages,anrTimeoutIntervalMillis
+             diagnosticLevel, cacheDirSize
+             */
+
+            options.beforeSend = { event in
+                self.setEventOriginTag(event: event)
+                /// TODO ? addPackages
+                return event
+            }
+            
+            self.options = options
+        }
+
+        result("iOS initNativeSdk" )
+    }
+
+    private func setEventOriginTag(event: Event){
+        guard let sdk = event.sdk else { return  }
+        if self.isValidSdk(sdk:sdk){
+
+            switch sdk["name"] as! String{
+            case "sentry.dart.flutter":
+                setEventEnvironmentTag(event:event,origin:"flutter",environment:"dart")
+            case "sentry.cocoa":
+                setEventEnvironmentTag(event:event,origin: "flutter",environment: "dart")
+            case "sentry.native":
+                setEventEnvironmentTag(event:event,origin: "flutter",environment: "dart")
+            default:
+                return
+            }
+        }
+    }
+
+    private func setEventEnvironmentTag(event: Event, origin: String = "ios", environment: String) {
+        event.tags?["event.origin"] = origin
+        event.tags?["event.environment"] = environment
+    }
+
+    private func isValidSdk( sdk: [String: Any]) -> Bool{
+        return (sdk["name"] != nil && !(sdk["name"] as! String).isEmpty)
+    }
+
+    func convertStringToDictionary(text: String) -> [String:AnyObject]? {
+        if let data = text.data(using: .utf8) {
+            do {
+                let json = try JSONSerialization.jsonObject(with: data, options: .mutableContainers) as? [String:AnyObject]
+                return json
+            } catch {
+                print("Something went wrong")
+            }
+        }
+        return nil
+    }
+
+    private func captureEnvelope(_ call: FlutterMethodCall, result: @escaping FlutterResult){
+        guard let arguments = call.arguments as? [Any],
+              !arguments.isEmpty,
+              let event = arguments.first as? String else {
+            result(FlutterError(code: "2", message: "Envelope is null or empty", details: nil) )
+            return
+        }
+
+
+        let parts = event.split(separator: "\n")
```

Comment:
maybe instead of passing a plain String from Dart to plugins, I could split up with a Map with 3 items, header, item header, and event, we can discuss this tomorow.

Reply:
It would be cleaner

Reply:
I'll think about it, cus then the parsing would be done on Android, so it's just choosing the side of the hardword.